pkg/docker/config/config.go

```diff
@@ -24,6 +24,7 @@ type dockerAuthConfig struct {
 type dockerConfigFile struct {
 	AuthConfigs map[string]dockerAuthConfig `json:"auths"`
 	CredHelpers map[string]string           `json:"credHelpers,omitempty"`
+	CredsStore  string                      `json:"credsStore,omitempty"`
```

Comment:
This looks to work for `docker login` but not for `podman login` or `buildah login` as the creds for those are stored in `$XDG_RUNTIME_DIR/containers/auth.json`?  Any chance to add checks for that file too?

Reply:
Sorry, that goes a little deeper than my familiarity — I don't use podman or buildah. But if the config files are separate won't they respect each files' contents — including `credsStore`?

Reply:
Oh, do you mean that `auths` can come from that other file when using those utilities? That sounds like another pull request to me, unrelated to this immediate change?

Reply:
Docker only knows about its auth file that lives in `~/.docker/config.json`.  Buildah and Podman first check `$XDG_RUNTIME_DIR/containers/auth.json` and if not found there, they then check the  `~/.docker/config.json` file.  So the contents are the same, the location differs.

I'd be fine with letting this in without changes for Podman/Buildah assuming a test or two is added.  However, if this goes through without the changes for Podman/Buildah I'd like to keep the issue open and have a note added about the other work still outstanding.

Reply:
The trouble is that looking at `modifyJSON`, edits only use `…/containers/auth.json`, and `~/.docker/config.json` is never consulted.

So, if the user sets up a `credsStore` in `~/.docker/config.json`, accessing credentials recorded in there will work fine util the first `podman login` for a registry — that will perform edits on `auth.json`, not see any `credsStore` configuration in there and just write into that file.

This is particularly painful because the default `…/containers/auth.json` locations are all temporary (per-user , cleared on reboot or even earlier), so it is not all that practical to configure `credsStore` in them.


OTOH, the current `credHelpers` code has exactly the same issue, so I’m fine with merging this PR without having to solve this here.

Reply:
Personally I don't use podman or buildah, I'm just using skopeo to supplement docker usage, so I don't have a `…/containers/auth.json`.